### PR TITLE
Create Python patch for Generativeai

### PR DESCRIPTION
**Pull request by psbang**
Create an import hook for the `google.generativeai` package so that upon import, the `configuration` method is overridden with Kaggle-specific headers and endpoints, and the `configuration` method is called once.

### CODE REVIEW

File: Dockerfile.tmpl [hunk 1]
```diff
@@ -578,7 +578,7 @@ RUN pip install pytorch-ignite \
         openpyxl \
         timm \
         torchinfo && \
-        pip install git+https://github.com/facebookresearch/segment-anything.git && \
```
Comment (Philmod) [inline Dockerfile.tmpl:581]: why did you remove the `&&`?

**Review by Philmod — COMMENTED** [review]
Could you add a test making sure the library works and if possible it respects your patch?

File: Dockerfile.tmpl (hunk 1 above)
Comment (psbang) [reply Dockerfile.tmpl:581]: Must've been a mistake

File: patches/generativeaipatch.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            default_metadata = []
+        kwargs['transport'] = 'rest' # For the initial launch of PaLM+Gemini, we only offer support for REST requests
+        default_metadata.append(("x-kaggle-proxy-data", os.environ['KAGGLE_DATA_PROXY_TOKEN']))
+        default_metadata.append(('x-kaggle-authorization', f'Bearer {os.environ['KAGGLE_USER_SECRETS_TOKEN']}'))
+        kwargs['default_metadata'] = default_metadata
+        if ('client_options' in kwargs):
+            client_options = kwargs['client_options']
+        else:
+            client_options = {}
+        client_options['api_endpoint'] = os.environ['KAGGLE_DATA_PROXY_URL'] + '/palmapi'
```
Comment (Philmod) [inline patches/generativeaipatch.py:20]: What about the other models? (watchout this is a public repo / pr)

Comment (psbang) [reply patches/generativeaipatch.py:20]: I'm not sure what you mean?

Comment (psbang) [reply patches/generativeaipatch.py:20]: All `generativeai` requests will be routed to the correct handler if the `palmapi` prefix is set.
